department-of-veterans-affairs/caseflow
# Justin/appeals 17533

Resolves [APPEALS-17533](https://vajira.max.gov/browse/APPEALS-17533)

# Description
- [ ] Moved intake edits from RadioField.jsx into new IntakeRadioField.jsx file
- [ ] Refactored/reset the RadioField.jsx file back to its initial state from the master branch
- [ ] Added a justification textbox to be rendered if a checkbox change is made 
- [ ] Saved Justification text into redux store to be utilized later
- [ ] Wrote and validated spec tests

## Acceptance Criteria
- [ ] AC 1. Create frontend logic for displaying the optional "why was this change made" text box when a user makes a change to the current status of the MST or PACT designation for an issue. 
- [ ] AC 2. User can include text in the textbox when editing the MST or PACT checkbox for a request issue, and Save the edited request  issue.
- [ ] AC 3. User can forego adding text in the textbox when editing the MST or PACT checkbox for a request issue, and Save the edited request  issue. 
- [ ] AC 4. SPEC test written
- [ ] AC 5. SPEC test validated

## Testing Plan
1. Log in as an intake user
2. Begin mail intake process
3. Choose VA Form 10182
4. Choose a veteran with an established appeal (for example: 400000021 or 400000025)
5. Select options on the review page and continue
6. Click on "Add Issue" button. The page should display something along the lines of: "Does issue 1 match any of these issues from past descriptions?"
7. Choose a past decision. The MST and PACT checkboxes should appear and allow the user to select them
8. When one or both of these checkboxes is selected, a new textbox should appear beneath these textboxes asking "Why was this change made?" This textbox should be optional to fill. 
9. The textbox will be saved to the redux state in the application but will not yet be visible within the application for this ticket. This functionality can be verified through the Redux Dev tools toolkit, inside appeal>addedIssues>0 (or whichever the last selected issue number was)>justification 

Screenshot included showing the state stored in redux tools:
![image](https://user-images.githubusercontent.com/95875751/235244607-bf56254b-2e43-4b1d-9d4c-9b784fe87e08.png)


# Frontend
## User Facing Changes

 BEFORE|AFTER
![image](https://user-images.githubusercontent.com/95875751/235243779-fbf11a78-060a-4991-812b-84aeee10da97.png)

 ---|---

![image](https://user-images.githubusercontent.com/95875751/235243832-25b6304d-6ed1-468e-84da-0623b6be2d12.png)
